src/mesh/mesh-amr_loadbalance.cpp

```diff
@@ -213,11 +214,13 @@ bool TryRecvFineToCoarse(int lid_recv, int send_rank, const LogicalLocation &fin
         // space if fine block is on the left side of a direction. I think this
         // should work fine even if the ownership model is changed elsewhere, since
         // the fine blocks should be consistent in their shared elements at this point
-        if (ox3 == 0) kb.e -= TopologicalOffsetK(te);
-        if (ox2 == 0) jb.e -= TopologicalOffsetJ(te);
+        if (ox3 == 0 && ndim > 2) kb.e -= TopologicalOffsetK(te);
+        if (ox2 == 0 && ndim > 1) jb.e -= TopologicalOffsetJ(te);
         if (ox1 == 0) ib.e -= TopologicalOffsetI(te);
-        const int ks = (ox3 == 0) ? 0 : (kb.e - kb.s + 1 - TopologicalOffsetK(te));
-        const int js = (ox2 == 0) ? 0 : (jb.e - jb.s + 1 - TopologicalOffsetJ(te));
+        const int ks =
+            (ox3 == 0 || ndim < 3) ? 0 : (kb.e - kb.s + 1 - TopologicalOffsetK(te));
+        const int js =
+            (ox2 == 0 || ndim < 2) ? 0 : (jb.e - jb.s + 1 - TopologicalOffsetJ(te));
```

Comment:
This fixes a bug which caused the z-component of coarsened face fields to be filled with zeros in 2D. I don't think anyone is likely to have run into it yet though (aside from in this example).